pkg/schema/definition/comment.go

```diff
@@ -0,0 +1,113 @@
+package definition
+
+import (
+	"encoding/json"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	blackfriday "github.com/russross/blackfriday/v2"
+)
+
+var (
+	regexpDefaults = regexp.MustCompile("(.*)Defaults to `(.*)`")
+	regexpExample  = regexp.MustCompile("(.*)For example: `(.*)`")
+	pTags          = regexp.MustCompile("(<p>)|(</p>)")
+
+	// patterns for enum-type values
+	enumValuePattern     = "^[ \t]*`(?P<name>[^`]+)`([ \t]*\\(default\\))?: .*$"
+	regexpEnumDefinition = regexp.MustCompile("(?m).*Valid [a-z]+ are((\\n" + enumValuePattern + ")*)")
```

Comment:
the code is now so different from the original skaffold code but this is the biggest thing I ended up leaving in.
Do we think it could be useful? Do we have/potentially have this type of enum in our config?
See [how it would look](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/schema/latest/config.go#L138-L143).

Reply:
Yes, I think the `nat` for example has only a few values it can take. There are probably a few more examples.